monai/data/image_writer.py

```diff
@@ -0,0 +1,395 @@
+        spatial_shape = metadata.get("spatial_shape")
+        return original_affine, affine, spatial_shape
+
+
+@require_pkg(pkg_name="itk")
+class ITKWriter(ImageWriter):
+    """
+    Write data and metadata into files on disk using ITK-python.
+    """
+
+    def __init__(self, output_dtype: DtypeLike = np.float32, **kwargs):
+        super().__init__(output_dtype=output_dtype, affine=None, channel_dim=0, **kwargs)
+
+    def set_data_array(self, data_array, channel_dim: Optional[int] = 0, squeeze_end_dims: bool = True, **kwargs):
+        """Convert ``data_array`` into 'channel-last' numpy ndarray."""
+        self.data_obj = self.convert_to_channel_last(
+            data=data_array,
+            channel_dim=channel_dim,
+            squeeze_end_dims=squeeze_end_dims,
+            spatial_ndim=kwargs.pop("spatial_ndim", 3),
+            contiguous=kwargs.pop("contiguous", True),
+        )
+        self.channel_dim = channel_dim
+
+    def set_metadata(self, meta_dict: Optional[Mapping] = None, resample: bool = True, **options):
+        """Resample ``self.dataobj`` if needed.  This method assumes ``self.data_obj`` is a 'channel-last' ndarray."""
+        original_affine, affine, spatial_shape = self.get_meta_info(meta_dict)
+        self.data_obj, self.affine = self.resample_if_needed(
+            data_array=self.data_obj,
+            affine=affine,
+            target_affine=original_affine if resample else None,
+            output_spatial_shape=spatial_shape,
+            mode=options.pop("mode", GridSampleMode.BILINEAR),
+            padding_mode=options.pop("padding_mode", GridSamplePadMode.BORDER),
+            align_corners=options.pop("align_corners", False),
+            dtype=options.pop("dtype", np.float64),
+        )
+
+    def write(self, filename_or_obj: PathLike, verbose: bool = False, **kwargs):
+        """Create an ITK object from ``self.data_obj`` and call ``itk.imwrite``"""
+        super().write(filename_or_obj, verbose=verbose)
+        self.data_obj = self.create_backend_obj(
+            self.data_obj, channel_dim=self.channel_dim, affine=self.affine, dtype=self.output_dtype, **kwargs  # type: ignore
+        )
+        itk.imwrite(
+            self.data_obj,
+            filename_or_obj,
+            compression=kwargs.pop("compression", False),
+            imageio=kwargs.pop("imageio", None),
+        )
+
+    @classmethod
+    def create_backend_obj(
+        cls,
+        data_array: NdarrayOrTensor,
+        channel_dim: Optional[int] = 0,
+        affine: Optional[NdarrayOrTensor] = None,
+        dtype: DtypeLike = np.float32,
+        **kwargs,
+    ):
+        """Create an ITK object from ``data_array``.  This method assumes a 'channel-last' ``data_array``."""
+        data_array = super().create_backend_obj(data_array)
+        _is_vec = channel_dim is not None
+        if _is_vec:
+            data_array = np.moveaxis(data_array, -1, 0)  # from channel last to channel first
```

Comment:
I think here you only revert the order of spatial-dims for ITK, it's not consistent with our `ITKReader` logic for **single channel data**, which reverts all the dims:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/image_reader.py#L346
Could you please help make them consistent in this PR?

Thanks.

Reply:
they are consistent here, it is in https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/image_reader.py#L348, there will be unit tests to check  `SaveImage`/`LoadImage` in the follow-up PRs.

Reply:
I mean for **single channel** data, would it be `img.GetNumberOfComponentsPerPixel() == 1`? So the logic will be different from here.

Thanks.